src/gt-channel.c

```diff
@@ -218,14 +222,33 @@ update_preview(GtChannel* self)
 
     if (priv->online)
     {
-        gt_twitch_download_picture_async(main_app->twitch, priv->preview_url, priv->cancel, 
-                                         (GAsyncReadyCallback) download_picture_cb, self); 
+        gt_twitch_download_picture_async(main_app->twitch, priv->preview_url, 0, priv->cancel,
+                                         (GAsyncReadyCallback) download_picture_cb, self);
     }
     else
     {
         if (priv->video_banner_url)
-            gt_twitch_download_picture_async(main_app->twitch, priv->video_banner_url, priv->cancel, 
-                                             (GAsyncReadyCallback) download_picture_cb, self); 
+        {
+            if (!g_file_test(priv->cache_filename, G_FILE_TEST_EXISTS))
+                g_info("{GtChannel} Cache miss for channel '%s'", priv->name);
+            else if (utils_timestamp_now() - utils_timestamp_file(priv->cache_filename) > 604800 + g_random_int_range(0, 604800))
```

Comment:
You are correct. However, this is not expected to be the final state of the code. At this time, the caching policy that was originally proposed is still in use for channel banners. The aim, of course, is to implement your suggestion.

I'm looking for feedback mainly on `gt-twitch.c`. Afterwards, I intend to apply the same logic to `gt-channel.c`